src/tunnel.cc

```diff
@@ -1101,15 +1101,13 @@ tunnelStart(ClientHttpRequest * http)
      * be allowed.  yuck, I know.
      */
 
-    if (Config.accessList.miss && !request->client_addr.isNoAddr()) {
+    if (Config.accessList.miss && !request->clientAddr().isNoAddr()) {
```

Comment:
Should we use the new `request->internal` here (and remove or adjust the above comment)? Or are there cases where internal (in the comment's sense) requests do not have `request->internal` set? Please double check whether additional polishing is needed to keep `request->internal` in sync with `request->clientAddr().isNoAddr()` and adjust the above source code comment as needed.

Reply:
> it looks like setNoAddr() is not called for source addresses for these internal requests and hence the existing check work incorrectly. Should we fix this inconsistency here, in this PR?

We probably should fix it (and it may be fixed automatically) _if_ we should modify this code to use our new HttpRequest::internal flag. We probably should fix it also if the fix is trivial and well-researched/understood (e.g., replacing isNoAddr() with a flag check). Otherwise, we probably should not fix this but may add a TODO or XXX, depending on your level of confidence that the code is buggy.

I think we should make the final decision _after_ addressing your second concern discussed below:

> Another question is [whether to] remove the new flag and use the old one instead

We should do that if the flags should represent the same concept. Otherwise, we should rename our new flag.

Please summarize the difference in how HttpRequest::internal and HttpRequest::flags::internal are defined/set/used today. If the two flags are nearly identical (bugs notwithstanding), then I am a bit surprised that you have not stumbled upon the existence of the old flag while adding all the code the sets and uses the new flag...

Reply:
> Please summarize the difference in how HttpRequest::internal and HttpRequest::flags::internal are defined/set/used today. 

* HttpRequest::internal is assigned by Squid instance itself when it creates HttpRequest object to send it by means of downloader or esi.
*  `HttpRequest::flags::internal` is assigned after receiving a request from another Squid instance if internalCheck() check returns true (the parsed path matched some hardcoded string in internal.cc).

(1) makes setNoAddr() call, so that the client ip will be logged as "-".
(2) does not make setNoAddr(), which is probably reasonable: we want to be able to log the client ip of the received 'internal' request.

It is not clear for me why Squid does not call setNoAddr() after creating 'internal' requests in contexts different from downloader and esi (e.g., peer_digest). Should not client ips for such requests be logged as "-" either? Probably these other requests (peer_digest, etc.) are not associated with ALE, but I am not sure.